pinot-server/src/main/java/org/apache/pinot/server/starter/helix/BaseServerStarter.java

```diff
@@ -374,6 +423,15 @@ public void start()
     LOGGER.info("Starting Pinot server");
     long startTimeMs = System.currentTimeMillis();
 
+    if (_serverConf.getProperty(Server.CONFIG_OF_STARTUP_ENABLE_TEMP_CLEANUP,
+        Server.DEFAULT_STARTUP_ENABLE_TEMP_CLEANUP)) {
+      File dataDir = new File(_serverConf.getProperty(CommonConstants.Server.CONFIG_OF_INSTANCE_DATA_DIR));
+      // We use 3 hours as the cutoff time as a general heuristic for when
```

Comment:
The lookback window can be encoded in the config?

Reply:
> Not a big fan of introducing a config just for cleaning up temp directory but I'm fine with having a fixed threshold just as you mentioned: delete for more than N hours. We shouldn't see multiple servers running on the same machine in produciton.

@jackjlli preferred it this way, and I'm inclined to agree. There's already a large number of configs, and ideally we remove this config in the future and make this default behavior on server startup.

Reply:
It'd be great to add more details on the purpose of introducing this threshold in the comment, like what we discuss in this PR.